### ISSUE TITLE
Generate config from template depending on available executables

### ISSUE THREAD

**Issue by ritzmann** [twpayne/chezmoi#798]
## What exactly are you trying to do?

My `.gitconfig` file currently looks like:
```
[core]
        pager = diff-so-fancy | less --tabs=4 -RFX
...
```

When I run `chezmoi apply`, I would like to detect if `diff-so-fancy` is installed and generate a different `pager` configuration if it is not installed.

## What have you tried so far?

I have converted `.gitconfig` to a template but I have not found a template function that would be able to find out if `diff-so-fancy` is installed, i.e. whether `diff-so-fancy` can be executed as a command.

I could set a variable in `chezmoi.toml` but that seems rather clunky.

## Where else have you checked for solutions?

* [x] I have read [chezmoi's how-to guide](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md), and not found the answer.
* [x] I have read [chezmoi's FAQ](https://github.com/twpayne/chezmoi/blob/master/docs/FAQ.md), and not found the answer.
* [x] I have searched [chezmoi's reference guide](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md), and not found the answer.
* [x] Other, please give details. Search text/template and Sprig documentation. Did an extensive web search.

## Output of any commands you've tried with `--verbose`

```
$ chezmoi command --verbose
```

## Output of `chezmoi doctor`

```
$ chezmoi doctor
     ok: version 1.8.2, commit 323cba56195179ade6b061fa2d6af37d16cd193e, built at 2020-06-09T17:52:08Z, built by homebrew
     ok: runtime.GOOS darwin, runtime.GOARCH amd64
     ok: /Users/user/.local/share/chezmoi (source directory, perm 700)
     ok: /Users/user (destination directory, perm 755)
     ok: /Users/user/.config/chezmoi/chezmoi.toml (configuration file)
     ok: /usr/local/bin/fish (shell)
     ok: /usr/local/bin/vi (editor)
     ok: /usr/local/bin/vimdiff (merge command)
     ok: /usr/bin/git (source VCS command, version 2.24.3)
     ok: /usr/local/bin/gpg (GnuPG, version 2.2.20)
warning: op (1Password CLI, not found)
warning: bw (Bitwarden CLI, not found)
warning: gopass (gopass CLI, not found)
     ok: /usr/local/bin/keepassxc-cli (KeePassXC CLI, version 2.5.4)
warning: lpass (LastPass CLI, not found)
warning: pass (pass CLI, not found)
warning: vault (Vault CLI, not found)
```

**Comment by twpayne** [comment]
Thank you for the clear request :)

#774 is relevant (cc @rajaravivarma-r)

It looks like there is sufficient demand for template functions that look for existing files and executables.

I'll add a couple of template functions to enable this.

**Comment by ritzmann** [comment]
I had actually started off with pyenv in my config.fish but then realized that is a bad example because I can always test if pyenv is executable with fish. 😸

**Comment by zb140** [comment]
Not to be too greedy, but since there are (or will be) new template functions that deal with file paths, would you consider adding something like `joinPath` that calls through to `filepath.Join` to make it easier for us Windows users to build platform independent paths to pass into those new functions?

**Comment by twpayne** [comment]
That's a great idea - I did realize that the `stat` example was a little OS-specific. I'll bundle it into #800 now.

**Comment by twpayne** [comment]
Fixed with #800 - please re-open if needed.

**Comment by m-rey** [comment]
Just wanted to mention that this use-case can be covered using `pager = (command -v diff-so-fancy >/dev/null && diff-so-fancy | less --tabs=4 -RFX || less) `  
That way, git would use `diff-so-fancy` if available, `less` otherwise.
That's how I have it currently in my setup, see https://github.com/m-rey/dotfiles/blob/54fe5568d4ba3c30d78ac6c99a7f68c631da8ab3/dot_config/git/config.tmpl#L38
